src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -703,6 +703,14 @@ private bool ShouldShareSymbol(ObjectNode node)
             if (!_targetPlatform.IsWindows)
                 return false;
 
+            // Types and methods from the compiler generated assembly are always shareable
+            CompilerGeneratedType type = (node is EETypeNode ? ((EETypeNode)node).Type : (node as MethodCodeNode)?.Method.OwningType) as CompilerGeneratedType;
```

Comment:
Consists of compiler generated types, of which `CompilerGeneratedType` is one example of. I should have named that type differently.

E.g. my pull request in flight (unboxing stubs) places `BoxedValueType` instances in the compiler generated assembly. It's not a problem for those because they will also be all generic, but an example of other random junk we might need to generate and fold.